src/battle_util.c

```diff
@@ -10803,6 +10791,30 @@ bool32 IsBattlerWeatherAffected(u8 battlerId, u32 weatherFlags)
     return FALSE;
 }
 
+// Utility Umbrella holders take normal damage from what would be rain- and sun-weakened attacks.
+u32 ApplyWeatherDamageMultiplier(u8 battlerAtk, u16 move, u8 moveType, u32 dmg, u16 holdEffectAtk, u16 holdEffectDef)
+{
+    u16 holdEffect;
+    if (WEATHER_HAS_EFFECT)
+    {
+        if ((gBattleWeather & B_WEATHER_RAIN) && holdEffectAtk != HOLD_EFFECT_UTILITY_UMBRELLA)
```

Comment:
This is still wrong. The target's Utility Umbrella is what alters weather damage, not the attacker's. The _only_ time the attacker's Utility Umbrella has an effect on weather damage is for Hydro Steam.
See the [Bulbapedia page](https://bulbapedia.bulbagarden.net/wiki/Utility_Umbrella):
[Water-type](https://bulbapedia.bulbagarden.net/wiki/Water_(type)) and [Fire-type](https://bulbapedia.bulbagarden.net/wiki/Fire_(type)) moves **targeted at the holder** will not have their power decreased or increased in rain or sun.

Reply:
Anubis (dataminer) stated otherwise though? ._.
https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9527435

Bulbapedia a lot of the time has gotter stuff wrong in the past, so I would take that with a grain of salt

Reply:
> Bulbapedia a lot of the time has gotten stuff wrong in the past, so I would take that with a grain of salt

So I did testing in SV because I agree it does seem weird, but it really is the target's Utility Umbrella that generally matters for weather damage. Hydro Steam is the weird exception that GF coded in to care about the attacker's Umbrella instead. Smogon's damage calculator also has it implemented this way, and what I saw there matched up with what I saw in game. What Anubis wrote about is specifically for Hydro Steam and not generic Fire-type moves in Sun.